lib/dotloop/admin.rb

```diff
@@ -1,13 +1,24 @@
 module Dotloop
   class Admin
+    include Dotloop::QueryParamHelpers
     attr_accessor :client
 
     def initialize(client:)
       @client = client
     end
 
-    def all(profile_id:)
-      @client.get("/profile/#{profile_id.to_i}/admin").map { |admin_attrs| Dotloop::Models::Admin.new(admin_attrs) }
+    def all(options = {})
+      admins = []
+      url = "/profile/#{profile_id(options)}/admin"
+      (1..MAX_LOOPS).each do |i|
+        options[:batch_number] = i
+        current_admin = @client.get(url, query_params(options)).map do |admin_attrs|
+          Dotloop::Models::Admin.new(admin_attrs)
+        end
+        admins += current_admin
+        break if current_admin.size < BATCH_SIZE
```

Comment:
You'll want to force the BATCH_SIZE in the options, or at least break on the options[:batch_size], otherwise the batch size won't match the options[:batch_size]